Framework/Core/src/ASoA.cxx

```diff
@@ -110,4 +110,11 @@ arrow::ChunkedArray* getIndexFromLabel(arrow::Table* table, const char* label)
   return table->column(index[0]).get();
 }
 
+void checkBinding(void const* binding, const char* tableName)
+{
+  if (O2_BUILTIN_UNLIKELY(binding == nullptr)) {
```

Comment:
The if statement (and only the `if`) should stay inlined, otherwise you will pay the cost of a function call.

Reply:
So only the `throw` should be extracted in a function? Would it have a benefit?

Reply:
Should I use `o2::framework::throw_error()` instead?

Reply:
You benefit the fact that the string / stack unrolling is not embedded in every single instance of that function.